### ISSUE TITLE
Error Schema refs need to be url-encoded

### ISSUE THREAD

**Issue by michelle-avery** [marshmallow-code/flask-smorest#136]
The automatic error response adding in .19 doesn't seem to be url-encoding the refs, so it's generating a schema like this:

```
post:
      responses:
        '202':
          description: The request was received
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/MySchema'
        '422':
          $ref: '#/components/responses/Unprocessable Entity'
        default:
          $ref: '#/components/responses/Default Error'
```

The Swagger editor, however throws an error that "$ref values must be RFC3986-compliant percent-encoded URIs" for any error with a space in it.

**Comment by lafrech** [comment]
A way to address this could be to change the error names to avoid having to url-encode.

`Default Error` is defined in flask-smorest. I can change it to `DefaultError` or `DEFAULT_ERROR`. In fact, I wrote `DefaultError` first then changed it for consistency with the other errors.

The other errors are defined in Python's `http.HTTPStatus` enum. We could use `name` attribute rather than `phrase` so that we'd have `UNPROCESSABLE_ENTITY`.

I'd prefer `UnprocessableEntity` but I found no straightforward way to get this and I'm reluctant to add more logic just for this cosmetic difference.

Well, we could do 

    string.capwords(status.name, '_').replace('_', '')

but still, isn't it better to stick to strings natively found un HTTPStatus? At least to help users wanting to access the values easily without any processing.
